bokeh/charts/_builder.py

```diff
@@ -106,21 +109,26 @@ class Builder(HasProps):
     xscale = String()
     yscale = String()
 
-    # Dimensional Modeling
+    # Dimensional Modeling (specified by Builder)
     dimensions = List(String, help="""The dimension
         labels that drive the position of the glyphs.""")
-    req_dimensions = Either(List(String), List(List(String)), List(Dict(String, String)), help="""The dimension
-        labels that must exist to produce the glyphs. This specifies what are the valid configurations
-        for the chart, with the option of specifying the type of the columns.""")
-
-    attributes = Dict(String, Instance(AttrSpec), help="""The attribute specs used to group data.""")
-    default_attributes = Dict(String, Instance(AttrSpec), help="""The attribute specs used to group data.""")
-
-    attribute_columns = List(ColumnLabel)
-
-    palette = Seq(Color, default=DEFAULT_PALETTE)
-
-    comp_glyphs = List(Instance(CompositeGlyph))
+    req_dimensions = Either(List(String), List(List(String)), List(Dict(String, String)),
+                            help="""The dimension labels that must exist to produce the
+                            glyphs. This specifies what are the valid configurations
+                            for the chart, with the option of specifying the type of the
+                            columns.""")
+
+    attributes = Dict(String, Instance(AttrSpec), help="""The attribute specs used to
+        group data.""")
+    default_attributes = Dict(String, Instance(AttrSpec), help="""The attribute specs
```

Comment:
I think we could improve the help text in order to make it clear about the difference between attributes and default_attributes